### PR TITLE
Chore: Refactor @someone

### LINKED ISSUE #430: Convert `@someone` to new event system

### PR DESCRIPTION
**Pull request by hasanlatch007**
!closes #430

### CODE REVIEW

File: src/programs/someone-tag.ts [hunk 1]
```diff
@@ -3,67 +3,76 @@ import Tools from "../common/tools";
...
 const QUESTION_SHEET_ID: string =
   "1eve4McRxECmH4dLWLJvHLr9fErBWcCGiH94ihBNzK_s";
 
-const someoneTag = async (message: Message) => {
-  const allow = await isAllowed(message.author);
+@Command({
+  event: DiscordEvent.MESSAGE,
+  trigger: "@someone",
+  channelNames: ["chat", "chat-too", "4th-chat", "chat-v"],
+  description: "This",
```
Comment (geisterfurz007) [inline src/programs/someone-tag.ts:15]:
```suggestion
  requiredRoles: ["Seek Discomfort"],
  description: "This",
```

File: src/programs/someone-tag.ts [hunk 2]
```diff
@@ -3,67 +3,76 @@ import Tools from "../common/tools";
...
-    const { member } = message;
-    const target = await getTarget(arg, message);
-    const question = await getQuestion();
-    if (!target)
-      await message.reply(
-        "There were no available users to ping! This is embarrassing. How could this have happened? There's so many people on here that statistically this message should never even show up. Oh well. Congratulations, I guess."
+    if (!hasSeekDiscomfort) {
+      await Tools.handleUserError(
+        message,
+        "You need the Seek Discomfort role for that! You can get one by writing a detailed bio of yourself in <#616616321089798145>."
```
Comment (geisterfurz007) [inline src/programs/someone-tag.ts:40]: This check can be ditched with `requiredRoles`. Less descriptive but the command is only used by people who know it anyway.

**Review by geisterfurz007 — CHANGES_REQUESTED** [review]

File: src/programs/someone-tag.ts (hunk 1 above)
Comment (hasanlatch007) [reply src/programs/someone-tag.ts]: Resolved with commit 82a9e7e

**Review by geisterfurz007 — APPROVED** [review]
🎃

**Pull request merged**